docs/AccessibleMenus.md

```diff
@@ -0,0 +1,112 @@
+## DOM structure
+The idea behind `src/js/common/utils/accessible-menus.js` is to be able to pass the DOM tree for a menu to the helper function `addMenuListeners` to make the menu keyboard navigable.
+
+The DOM structure needed on the menu for the helper function to work properly is as follows.
+
+```html
+<!--
+  The role of the root element can be either "menu" or "menubar"
+-->
+<ul role="[menu|menubar]">
+  <li>
+    <menu/> <!-- See below -->
+  </li>
+  <!-- Or -->
+  <li>
+    <a> <!-- Any menu item -->
+  </li>
+</ul>
+```
+
+Where `<menu/>` has the following structure:
+```html
+<!--
+  When the menu is expanded, the button gets aria-expanded="true"
+  When it's collapsed, the attribute is removed
+-->
+<!--
+  Menu button can either be a button or have role="button"
+  Menu button's aria-controls points to either the menu's id or its parent's id
+-->
+<button
+  role="button"
+  aria-haspopup="true"
+  aria-controls="foo"/>
+<!-- Optional div nesting -->
+<div id="foo" hidden>
+  <!--
+    The ul has id="foo" and hidden only if its parent doesn't have id="foo"
+  -->
+  <ul id="foo" role="menu" hidden>
+    <li>
+      <menu/> <!-- Nested menus possible -->
+    </li>
+    <!-- Or -->
+    <li>
+      <a> <!-- Any menu item -->
+    </li>
+  </ul>
+</div>
+```
+
+
+## Functionality
+A user should be able to use the keyboard to do the following things:
+
+*Menubar*
+1. Navigate between visible menubar items
+  - Left / right arrow keys
+  - With wraparound
+2. Open sub menus
+  - Down arrow / space / enter
+  - Focus should go to the first visible menu item in the sub-menu
+3. Activate menubar menu items
+  - Follow links with space / enter
+
+*Menus*
+The helper function is pretty naive right now, but it can be modified to be smarter. Currently, it assumes menus are vertical and sub-menus are opened to the right.
```

Comment:
Would it be of interest to create ticket for planned optimizations/list them here?

Reply:
Mmm...I'm not sure, actually. For this particular enhancement, I think just mentioning this assumption is enough for now. For others... :man_shrugging: Probably a case-by-case basis.

I think planning for work is best recorded in separate tickets, though, not documentation.